client/components/gridicon/index.tsx

```diff
@@ -6,41 +6,38 @@ import {
 } from 'gridicons/dist/util/icons-offset';
 import spritePath from 'gridicons/svg-sprite/gridicons.svg';
 import React from 'react';
-import { Assign } from 'utility-types';
-
-function needsOffset( name: string, icons: string[] ) {
-	return icons.indexOf( name ) >= 0;
-}
+import type { ReactNode } from 'react';
+import type { Assign } from 'utility-types';
 
 interface Props {
 	icon: string;
 	size?: number;
+	title?: ReactNode;
 }
+
 type AllProps = Assign< React.SVGProps< SVGSVGElement >, Props >;
 
 const Gridicon = React.forwardRef< SVGSVGElement, AllProps >( ( props: AllProps, ref ) => {
-	const { size = 24, icon, onClick, className, title, ...otherProps } = props;
+	const { size = 24, icon, className, title, ...otherProps } = props;
 	const isModulo18 = size % 18 === 0;
 
 	// Using a missing icon doesn't produce any errors, just a blank icon, which is the exact intended behaviour.
 	// This means we don't need to perform any checks on the icon name.
 	const iconName = `gridicons-${ icon }`;
-	const offsetClasses = isModulo18
-		? [
-				needsOffset( iconName, iconsThatNeedOffset ) ? 'needs-offset' : false,
-				needsOffset( iconName, iconsThatNeedOffsetX ) ? 'needs-offset-x' : false,
-				needsOffset( iconName, iconsThatNeedOffsetY ) ? 'needs-offset-y' : false,
-		  ]
-		: [];
-	const iconClass = classnames( 'gridicon', iconName, className, ...offsetClasses );
+
+	const iconClass = classnames( 'gridicon', iconName, className, {
+		'needs-offset': isModulo18 && iconsThatNeedOffset.includes( iconName ),
+		'needs-offset-x': isModulo18 && iconsThatNeedOffsetX.includes( iconName ),
+		'needs-offset-y': isModulo18 && iconsThatNeedOffsetY.includes( iconName ),
+	} );
+
 	return (
 		<svg
 			xmlns="http://www.w3.org/2000/svg"
 			viewBox="0 0 24 24"
 			className={ iconClass }
 			height={ size }
 			width={ size }
-			onClick={ onClick }
```

Comment:
I'm guessing this was removed simply because it's not being used anywhere?

Reply:
The `onClick` prop is passed as part of `...otherProps`. In other words, I changed
```js
const { icon, onClick, ...otherProps } = props;
return <svg onClick={ onClick } { ...otherProps } />;
```
to
```js
const { icon, ...otherProps } = props;
return <svg { ...otherProps } />;
```

Reply:
Good point, I missed that 👍